src/builders/enzymes.jl

```diff
@@ -113,3 +131,217 @@ function gene_product_isozyme_constraints(
+all the fluxes in `reaction_isozymes` (mass/gDW units). Alternatively,
+`capacity` may be a vector of identifier-fluxes-limit triples that make a
+constraint (identified by the given identifier) that limits the enzyme
+requirements of the listed fluxes to the given limit (mass/gDW units).
+
+Note: [`simplified_enzyme_constraints`](@ref) and [`enzyme_constraints`](@ref)
+differ in how the capacity bound(s) are formulated. For the former, fluxes are
+used, but for the latter, gene products are used directly.
+"""
+function simplified_enzyme_constraints(
+    constraints::C.ConstraintTree;
+    fluxes = constraints.fluxes,
+    fluxes_reverse = constraints.fluxes_reverse,
+    fluxes_forward = constraints.fluxes_forward,
+    reaction_isozymes::Dict{String,Dict{String,Isozyme}},
+    gene_product_molar_masses::Dict{String,Float64},
+    capacity::Union{Vector{Tuple{String,Vector{String},Float64}},Float64},
+)
+    # get fastest isozyme for each reaction (flux * MW/kcat = isozyme mass required)
+    enzyme_speeds = Dict(
+        Symbol(rid) => (;
+            forward = minimum(
+                sum(
+                    stoich * gene_product_molar_masses[gid] for
+                    (gid, stoich) in iso.gene_product_stoichiometry
+                ) / iso.kcat_forward for iso in values(isos)
+            ),
+            reverse = minimum(
+                sum(
+                    stoich * gene_product_molar_masses[gid] for
+                    (gid, stoich) in iso.gene_product_stoichiometry
+                ) / iso.kcat_reverse for iso in values(isos)
+            ),
+        ) for (rid, isos) in reaction_isozymes
+    )
+
+
+    # connect all parts with constraints
+    :directional_flux_balance^sign_split_constraints(
+        positive = fluxes_forward,
+        negative = fluxes_reverse,
+        signed = fluxes,
+    ) *
+    :capacity^(
+        capacity isa Float64 ?
+        C.Constraint(
+            value = sum(
+                C.value(c) * enzyme_speeds[rid].forward for
+                (rid, c) in fluxes_forward if haskey(enzyme_speeds, rid)
+            ) + sum(
+                C.value(c) * enzyme_speeds[rid].reverse for
+                (rid, c) in fluxes_reverse if haskey(enzyme_speeds, rid)
+            ),
+            bound = C.Between(0, capacity),
+        ) :
+        C.ConstraintTree(
+            Symbol(id) => C.Constraint(
+                value = sum(
+                    C.value(get(fluxes_forward, rid, 0)) *
+                    enzyme_speeds[rid].forward for
+                    rid in flxs if haskey(enzyme_speeds, rid)
+                ) + sum(
+                    C.value(get(fluxes_reverse, rid´, 0)) *
+                    enzyme_speeds[rid].reverse for
+                    (rid, c) in flxs if haskey(enzyme_speeds, rid)
+                ),
+                bound = C.Between(0, limit),
+            ) for (id, flxs, limit) in capacity_limits
+        )
```

Comment:
lots of shared code here, maybe we could have a common helper

Reply:
yeah, some shared code